### ISSUE TITLE
Generation of type parameters with single constraints

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe#5072]
Another consolidation issue:

#2548
#2712
#4985
#5026

**Comment by Simn** [comment]
If I understand @hughsando correctly he wants a pre-generation filter which replaces all occurrences of `T:C` with `C`.

While this makes sense, it's not entirely trivial to implement. One concern is that we'll have to expression-map everything because the field access on `TField` would change.

I don't think this is something I could rush into 3.3 with a good conscience, so it might still be necessary to deal with this at gencpp-level for the time being. We should at least avoid the regressions (#4985 and #5026) here.

**Comment by hughsando** [comment]
Yes, that sounds like a good plan.  I will do dynamic access now, and then the haxe fairies will come along and make it run faster without me doing anything different, since it will use a different code path.

**Comment by Simn** [comment]
I'm assigning it to you to deal with the C++ issue itself.

**Comment by hughsando** [comment]
These ones to do with "unknown object type..." have been solved by:
https://github.com/HaxeFoundation/haxe/commit/5f25fffe5e7e3780c64c1410457110d01d84e736

The template efficiency issues have been solved by the way interfaces now work.  I guess there might be a class efficiency issue which has not been addressed.
